docs/src/pages/style/color/ColorTool.js

```diff
@@ -0,0 +1,282 @@
+  'orange',
+  'deepOrange',
+];
+
+const shades = [900, 800, 700, 600, 500, 400, 300, 200, 100, 50, 'A700', 'A400', 'A200', 'A100'];
+
+export const styles = theme => ({
+  root: {
+    display: 'flex',
+    width: '100%',
+  },
+  radio: {
+    width: 47,
+    height: 47,
+  },
+  radioSelected: {
+    width: 47,
+    height: 47,
+    border: '1px solid white',
+    color: theme.palette.common.white,
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  sliderContainer: {
+    display: 'flex',
+    justifyContent: 'spece-between',
+    width: 192,
+    marginTop: theme.spacing.unit * 2,
+    marginBottom: theme.spacing.unit,
+  },
+  sliderTypography: {
+    marginTop: 5,
+  },
+  swatch: {
+    width: 192,
+    backgroundColor: theme.palette.common.white,
+  },
+  input: {
+    marginTop: 24,
+    width: 192,
+  },
+  button: {
+    marginTop: 24,
+  },
+});
+
+class ColorTool extends React.Component {
+  state = {
+    primary: '#2196f3',
+    secondary: '#f50057',
+    primaryText: '#2196f3',
+    secondaryText: '#f50057',
+    primaryHue: 'blue',
+    secondaryHue: 'pink',
+    primaryShade: 4,
+    secondaryShade: 11,
+  };
+
+  hashPrefix = string => (string.charAt(0) === '#' ? string : `#${string}`);
+
+  isRgb = string => /#?([0-9a-f]{6})/i.test(string);
+
+  handleChangeColor = name => event => {
+    const {
+      target: { value: color },
+    } = event;
+
+    this.setState({
+      [`${name}Text`]: color,
+    });
+
+    if (this.isRgb(color)) {
+      this.setState({
+        [name]: color,
+      });
+    }
+  };
+
+  handleChangeHue = name => event => {
+    const {
+      target: { value: hue },
+    } = event;
+
+    this.setState(currentState => {
+      const color = colors[hue][shades[currentState[`${name}Shade`]]];
+      return {
+        [`${name}Hue`]: hue,
+        [name]: color,
+        [`${name}Text`]: color,
+      };
+    });
+  };
+
+  handleChangeShade = name => (event, shade) => {
+    this.setState(currentState => {
+      const color = colors[currentState[`${name}Hue`]][shades[shade]];
+      return {
+        [`${name}Shade`]: shade,
+        [name]: color,
+        [`${name}Text`]: color,
+      };
+    });
+  };
+
+  handleChangeDocsColors = values => {
+    const primaryMain = values.hasOwnProperty('primary') ? values.primary : this.state.primary;
+    const secondaryMain = values.hasOwnProperty('secondary')
+      ? values.secondary
+      : this.state.secondary;
+
+    this.props.dispatch({
+      type: actionTypes.THEME_CHANGE_PALETTE_COLORS,
+      payload: {
+        paletteColors: {
+          primary: { main: primaryMain },
+          secondary: { main: secondaryMain },
+        },
+      },
+    });
+  };
+
+  render() {
+    const { classes } = this.props;
+    const {
+      primary,
+      secondary,
+      primaryText,
+      secondaryText,
+      primaryShade,
+      secondaryShade,
+    } = this.state;
+
+    const ColorTile = (hue, colorIntent) => {
```

Comment:
Can this component be moved outside of the render method? (simpler I guess)

Reply:
Oh no, it's not a component, it's a function: `colorTile`?

Reply:
Ah, right - I shorted it from `getColorTile` and missed the capitalisation.